src/Symfony/Component/Yaml/Parser.php

```diff
@@ -300,7 +303,7 @@ private function doParse(string $value, int $flags)
                 $subTag = null;
                 if ($mergeNode) {
                     // Merge keys
-                } elseif (!isset($values['value']) || '' === $values['value'] || 0 === strpos($values['value'], '#') || (null !== $subTag = $this->getLineTag($values['value'], $flags)) || '<<' === $key) {
+                } elseif (!isset($values['value']) || '' === $values['value'] || 0 === strncmp($values['value'], '#', 1) || (null !== $subTag = $this->getLineTag($values['value'], $flags)) || '<<' === $key) {
```

Comment:
Could be replaced with `'#' === $values['value'][0]` which is safe because the case will not be entered if string offset `0` does not exist. I'll make this change.